### PR TITLE
introduce ExceptionDummyPollDecorator

### PR DESCRIPTION
**Pull request by d-rothe**
extend ExceptionDummyBackend so that all poll-type registers are
decorated with ExceptionDummyPollDecorator which allows to set data
validity of individual registers to invalid

### CODE REVIEW

File: device_backends/include/ExceptionDummyBackend.h [hunk 1]
```diff
@@ -296,6 +320,47 @@ namespace ChimeraTK {
...
+      _target->preRead(type);
+    }
+
+    void doPostRead(TransferType type, bool updateDataBuffer) override {
+      NDRegisterAccessorDecorator<UserType, UserType>::doPostRead(type, updateDataBuffer);
+      // overwriting is only allowed for faulty.
+      if(_backend->getValidity(_path) == DataValidity::faulty) this->_dataValidity = DataValidity::faulty;
+    }
+
+    bool _makeInvalid = true;
```
Comment (mhier) [inline device_backends/include/ExceptionDummyBackend.h:353]: is this used anywhere?

**Review by mhier — CHANGES_REQUESTED** [review]

File: device_backends/include/ExceptionDummyBackend.h (hunk 1 above)
Comment (d-rothe) [reply device_backends/include/ExceptionDummyBackend.h:353]: No longer. Removed it.

File: device_backends/include/ExceptionDummyBackend.h [hunk 2]
```diff
@@ -164,6 +174,20 @@ namespace ChimeraTK {
     /// Protected by _pushDecoratorMutex
     bool _activateNewPushAccessors{false};
 
+    std::mutex _registerValiditiesMutex;
+    std::map<RegisterPath, DataValidity> _registerValidities;
+    void setValidity(RegisterPath path, DataValidity val) {
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply device_backends/include/ExceptionDummyBackend.h:179]: Done

File: device_backends/include/ExceptionDummyBackend.h [hunk 3]
```diff
@@ -296,6 +320,47 @@ namespace ChimeraTK {
...
+    ExceptionDummyPollDecorator(const boost::shared_ptr<ChimeraTK::NDRegisterAccessor<UserType>>& target,
+        const boost::shared_ptr<ExceptionDummy>& backend)
+    : NDRegisterAccessorDecorator<UserType>(target), _backend(backend) {
+      assert(_target->isReadable());
+
+      _path = _target->getName();
+      _path.setAltSeparator(".");
+    }
+
+    void doPreRead(TransferType type) override {
```
[the remark this answers is not in the document]
Comment (d-rothe) [reply device_backends/include/ExceptionDummyBackend.h:340]: removed

**Review by mhier — APPROVED** [review]

**Pull request merged**